cmd/juju/commands/upgradejuju.go

```diff
@@ -652,3 +652,9 @@ func makeUploadVersion(vers version.Number, existing coretools.List) version.Num
 	}
 	return vers
 }
+
+func compareNoBuild(a, b version.Number) int {
+	a.Build = 0
+	b.Build = 0
```

Comment:
Maybe rename this to compareMajorMinor and only compare those two?

The *really* important thing to me is that the check should be compatible with whatever our "upgrade" logic is going to accept once the request has actually gone through.